pkg/status/controller.go

```diff
@@ -203,6 +206,8 @@ func (c *Controller) run(ctx context.Context, workers int, cListers chan interfa
 	}) // this will have the broker call the callback for all existing resolutions
 	c.logger.Info("Registered binding resolution broker callback")
 
+	c.buildSingletonStateAndOptionallyReconcile(ctx, false)
```

Comment:
Why does this need to be done now? If I understand correctly, all this does is populate an internal data structure that is later consulted on a workload object-by-object basis. Why not just determine the needed bit when syncing that workload object?

Reply:
This data structure is keeping track of one bit per workload object, indicating whether singleton status return should be done for that object. That is the "needed bit" that I meant.

Reply:
I see what you meant. To get the value of the that bit for _one_ workload object, the reconciliation needs to go over _all_ Binding objects. So I thought it is more efficient to do a full build (for _all_ workload object together) here, then do incremental updates in reconciliations. That's why I had this line here.

But right now I don't yet have an idea on how to do the incremental updates. Because an event of _one_ Binding object is not always sufficient to compute the incremental. For example, if a Binding object says "I don't select this workload object anymore", can I mark the corresponding bit to `false`? No, because other Binding object(s) may still cover the workload object, I still have to go over all the Binding objects to determine that bit. To me it seems that a full build every time is necessary so I removed the `TODO`.